ARExperiment/ARExperiment/LightsAnimationsViewController.swift

```diff
@@ -40,30 +28,17 @@ class LightsAnimationsViewController: UIViewController {
         sceneView.session.pause()
     }
 
-    private func setUpModelsOnLoad() {
-        earthNode = modelForNodeName(earthNodeName)
-        moonNode = modelForNodeName(moonNodeName)
-        planeNodeModel = modelForNodeName(planeNode)
-        lightNodeModel = modelForNodeName(lightNode)
-    }
-
-    private func modelForNodeName(_ nodeName: String) -> SCNNode? {
-        return arModel.createSceneNodeForAsset(nodeName,
-                                               assetFolder: assetFolder,
-                                               fileName: fileName,
-                                               assetExtension: fileExtension)
-    }
-
     override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
         guard let location = touches.first?.location(in: sceneView) else {
             return
         }
 
-        if let nodeExists = sceneView.scene.rootNode.childNode(withName: earthNodeName, recursively: true),
-            let secondNodeExists = sceneView.scene.rootNode.childNode(withName: moonNodeName, recursively: true) {
-            nodeExists.removeFromParentNode()
-            secondNodeExists.removeFromParentNode()
+        for node in ARAsset.earthMoon.nodes {
```

Comment:
extract the asset in a separate var, so that here you don't specify "eartMoon". Here we iterate over all nodes from an asset, not the "earthMoon's assets" specifically.